p2p/types/src/peer_id.rs

```diff
@@ -41,6 +41,10 @@ impl PeerId {
         let id = NEXT_PEER_ID.fetch_add(1, Ordering::Relaxed);
         Self(id)
     }
+
+    pub fn from_u64(n: u64) -> Self {
```

Comment:
This is kind of nit-picky, but if you just need this for tests to initialize a random value, you can have a function like `random_using(...)`

Reply:
The issue is that I use this in a simulation test that performs all sorts of random operations on the orphan storage data structure. To ensure good coverage, `PeerId`s are generated from a carefully selected random distribution. That distribution may not be suitable for other use cases, testing or otherwise, so not a good idea to move the generation here IMO.

Reply:
Alrighty. No worries.